### PR TITLE
Prevent compiler warnings related to abnormal exit and fallthrough.

### PR DESCRIPTION
**Pull request by eric-s-raymond**
This PR was motivated by a failed attempt to build go-interpreter/chezgo, which sets -Werrors..

It fixes several trivial warnings.  There are a few others that look nontrivial, which I will investigate further.

No logic changes.

### CODE REVIEW

**Review by burgerrg — COMMENTED** [review]
Thanks for this change. Please put the pacify comment on every unneeded break, and add a line to LOG.

**Comment by eric-s-raymond** [comment]
Done.

With this change you're pretty close to compiling clean with -Werrors.  I recommend pushing it the rest of the way.

**Comment by jltaylor-us** [comment]
Not that this should prevent us from merging this PR, but I thought we already built with `-Werror`...

On macOS, we use `-Wall -Wextra -Werror -Wno-implicit-fallthrough`, and from a quick glance it looks like the same in the linux makefiles.  This PR will allow us to remove the `-Wno-implicit-fallthrough` – I've verified that make in `a6osx/c` succeeds after removing the flag (on macOS, obviously); other builds would have to be verified (but maybe we can assume the GitHub runners will cover it).

It looks like go-interpreter/chezgo is using a copy of the code already configured for a6osx from 6 years ago, which is before the a6osx makefile was updated to add `-Wno-implicit-fallthrough` (see #273).

Are there any _other_ errors preventing a clean compile with `-Werror` with some other OS/compiler combination?

File: LOG [hunk 1]
```diff
@@ -2373,3 +2373,5 @@
 - fix repair to invalid live-pointer mask, where a pass did not handle the
   new raw-data form correctly
     mats/fx.ms s/cpnanopass.ss
+- avoid sime trivial warnings related to abnormal exit and fallthrough.
```
Comment (burgerrg) [inline LOG:2376]: `sime trivial` => `some false-positive`

**Review by burgerrg — COMMENTED** [review]
Thanks! Please remove the -Wno-implicit-fallthrough since it shouldn't be required any more.

**Review by burgerrg — APPROVED** [review]

**Pull request merged**